docs/metrics-yaml-format.md

```diff
@@ -0,0 +1,254 @@
+manipulate the values of metrics before sending to the receiver. Some of the use cases are to extract a number from a string 
+value or derive a metric as a function of 2 attributes. 
+
+For derived metrics, the `source` field of the metric has to be of the format: `func:<function-name>(<params>)`. 
+
+* `func`: Denotes that metric has to be derived by applying the function specified.
+* `<function-name>`: Function to invoked. In the case of [built-in functions](./built-in-functions.md), 
+just function name is enough. For custom functions, specify the function name with fully qualified classname. e.g.
+`func:com.sematext.spm.client.solr.CalculateWarmupTime(warmupTime,searcherName)`. The function class should extend 
+`com.sematext.spm.client.observation.CalculationFunction`
+* `<params>`: Optional params to function. All params are of type String and will be interpreted and converted 
+accordingly in the function implementation.
+
+By default, the result of the function will be interpreted as a gauge. If the result is counter, then prepend the function
+reference with `as_counter`. For example,
+
+```yaml
+  - name: cache.lookups
+    source: as_counter:func:ExtractLongFromMapString(Value,cumulative_lookups)
+    type: counter
+    label: cache lookups
+    description: lookups count
+    stateful: true
+```
+
+Some of the built-in functions can be applied to tags also. 
+
+## Percentiles
+
+Sematext App Agent, can automatically calculate percentiles for a metric and send to the receiver. The percentiles to be
+calculated can be specified using `pctls` field. Below example will calculate 99th, 95th and 50th percentile of metric 
+`requestCount` and send it in attributes `request.count.pctl.99`, `request.count.pctl.95` and `request.count.pctl.50` 
+respectively.
+
+```yaml
+  - name: request.count
+    source: requestCount
+    type: double_gauge
+    stateful: true
+    pctls: 99,95,50
+    label: request count
+```
+
+## DB Vertical Model
+In the case of DB source, the metrics to be collected can be present in a single row or in multiple rows, 
+with each row containing the metric name and value. In such cases, you can set `dbVerticalModel` to `true`. For example,
+for the table below, `dbVerticalModel` will be set to true.
+```
+┌─event───────────────────────────────────┬─────────value─┐
+│ Query                                   │        335485 │
+│ SelectQuery                             │        335485 │
+│ FileOpen                                │           149 │
+│ ReadBufferFromFileDescriptorRead        │           257 │
+│ ReadBufferFromFileDescriptorReadBytes   │         23956 │
+│ WriteBufferFromFileDescriptorWrite      │             1 │
+│ WriteBufferFromFileDescriptorWriteBytes │            59 │
+```
+
+## Specifying Variables in YAML
+The App Agent allows specifying set up time variables in YAML. The variable names can be referred in the values using 
```

Comment:
The App Agent allows specifying set up time variables in YAML => The App Agent lets you specify variables in YAML
The variable names can be referred => The variable names can be referenced